promql/testdata/native_histograms.test

```diff
@@ -0,0 +1,207 @@
+	{} 0.5
+
+# We expect all values to fall in the range 0 < x <= 8.
+eval instant at 5m histogram_fraction(0, 8, single_histogram{})
+	{} 1
+
+# Median is 1.5 due to linear estimation of the midpoint of the middle bucket, whose values are within range 1 < x <= 2.
+eval instant at 5m histogram_quantile(0.5, single_histogram{})
+	{} 1.5
+
+
+
+# Repeat the same histogram 10 times.
+load 5m
+	multi_histogram{}	{{schema:0 sum:4 count:4 buckets:[1 2 1]}}x10
+
+eval instant at 5m histogram_count(multi_histogram{})
+	{} 4
+
+eval instant at 5m histogram_sum(multi_histogram{})
+	{} 4
+
+eval instant at 5m histogram_fraction(1, 2, multi_histogram{})
+	{} 0.5
+
+eval instant at 5m histogram_quantile(0.5, multi_histogram{})
+	{} 1.5
+
+
+# Each entry should look the same as the first.
+eval instant at 50m histogram_count(multi_histogram{})
+	{} 4
+
+eval instant at 50m histogram_sum(multi_histogram{})
+	{} 4
+
+eval instant at 50m histogram_fraction(1, 2, multi_histogram{})
+	{} 0.5
+
+eval instant at 50m histogram_quantile(0.5, multi_histogram{})
+	{} 1.5
+
+
+
+# Accumulate the histogram addition for 10 iterations, offset is a bucket position where offset:0 is always the bucket
+# with an upper limit of 1 and offset:1 is the bucket which follows to the right. Negative offsets represent bucket
+# positions for upper limits <1 (tending toward zero), where offset:-1 is the bucket to the left of offset:0.
+load 5m
+	incr_histogram{}	{{schema:0 sum:4 count:4 buckets:[1 2 1]}}+{{sum:2 count:1 buckets:[1] offset:1}}x10
+
+eval instant at 5m histogram_count(incr_histogram{})
+	{} 5
+
+eval instant at 5m histogram_sum(incr_histogram{})
+	{} 6
+
+# We expect 3/5ths of the values to fall in the range 1 < x <= 2.
+eval instant at 5m histogram_fraction(1, 2, incr_histogram{})
+	{} 0.6
+
+eval instant at 5m histogram_quantile(0.5, incr_histogram{})
+	{} 1.5
+
+
+eval instant at 50m histogram_count(incr_histogram{})
+	{} 14
+
+eval instant at 50m histogram_sum(incr_histogram{})
+	{} 24
+
+# We expect 12/14ths of the values to fall in the range 1 < x <= 2.
+eval instant at 50m histogram_fraction(1, 2, incr_histogram{})
+	{} 0.8571428571428571
+
+eval instant at 50m histogram_quantile(0.5, incr_histogram{})
+	{} 1.5
+
+
+
+# Schema represents the histogram resolution, different schema have compatible bucket boundaries, e.g.:
+#  0: 1 2 4 8 16 32 64 (higher resolution)
+# -1: 1   4   16    64  (lower resolution)
+#
+# Histograms can be merged as long as the histogram to the right is same resolution or higher.
+load 5m
+	low_res_histogram{}	{{schema:-1 sum:4 count:1 buckets:[1] offset:1}}+{{schema:0 sum:4 count:4 buckets:[2 2] offset:1}}x1
+
+eval instant at 5m histogram_count(low_res_histogram{})
+	{} 5
+
+eval instant at 5m histogram_sum(low_res_histogram{})
+	{} 8
+
+# We expect all values to fall into the lower-resolution bucket with the range 1 < x <= 4.
+eval instant at 5m histogram_fraction(1, 4, low_res_histogram{})
+	{} 1
```

Comment:
Same as before, let's test the full histogram.

Reply:
Added in 3eb0735a7, see above regarding parser support.

Reply:
Fixed in 65cbd1932